### PR TITLE
Fix excessive eval review queue routing

### LINKED ISSUE #85: Reduce excessive needs-human-review routing in eval scoring
## Problem

Evaluation currently routes too many plausible findings into the needs-human-review queue, making automated benchmark metrics uninformative.

A recent private benchmark cohort produced this aggregate outcome:

- Seven completed rows
- Six non-empty reports containing 61 candidate findings
- All 61 candidates routed to needs-human-review
- Zero candidates counted as true positives or false positives
- Precision, recall, and F1 therefore all reported as zero

No benchmark-specific project or finding details are included here.

Two evaluator behaviors appear to combine into this failure mode:

1. Terminal reports may satisfy the broad report contract while their issue objects do not satisfy the normalized finding schema. Evaluation still scores those issue objects directly, so structured matching signals can be absent even though the report was accepted upstream.
2. Once the deterministic matcher classifies a plausible candidate as needs-human-review, normalizeLlmJudgeResult preserves that classification. The LLM judge cannot promote it to a true positive, even when it identifies a reference candidate and returns a score at or above the configured threshold.

As a result, the review queue becomes the default destination for plausible findings rather than a narrow path for strong novel findings.

## Desired behavior

- A credible partial match on root cause should count as a valid finding when the affected behavior and impact are consistent, even if localization or reproduction evidence is incomplete.
- The configured LLM judge should be allowed to promote a deterministic review candidate to true-positive when it selects a valid reference candidate and meets the score/confidence policy.
- needs-human-review should be reserved for strong, well-supported novel findings that do not match the reference corpus.
- Weak or unsupported unmatched findings should be classified as false-positive rather than filling the review queue.
- Terminal report issues should be normalized into the canonical finding schema before matching, or evaluation should consume a canonical normalized-findings artifact directly.

## Suggested classification policy

| Candidate | Expected classification |
| --- | --- |
| Strong deterministic match | true-positive |
| Credible partial root-cause match confirmed by the judge | true-positive |
| Strong, substantiated, apparently novel finding | needs-human-review |
| Weak, vague, or unsupported unmatched finding | false-positive |
| Malformed report data | explicit validation failure, not silent zero metrics |

## Relevant code

- packages/evals/src/scoring.ts: scoreFindings and bestMatch
- packages/evals/src/scoring.ts: normalizeLlmJudgeResult
- packages/evals/src/scoring.ts: readReport
- packages/artifacts/src/findings-schema.ts: canonical normalized finding schema
- packages/artifacts/src/artifact-contracts.ts: terminal report contract

## Acceptance criteria

- [ ] Report issue objects are normalized or rejected before matching; accepted reports cannot silently lose the fields required by evaluation.
- [ ] A deterministic needs-human-review result can be promoted by the configured judge when it selects a valid reference candidate and satisfies the threshold policy.
- [ ] Partial root-cause matches have an explicit, tested promotion policy.
- [ ] Strong novel findings remain queued for review.
- [ ] Weak unmatched findings are classified as false-positive.
- [ ] Scoring emits reason codes or aggregate counters explaining why findings entered the review queue.
- [ ] Regression tests cover deterministic matches, judge-confirmed partial matches, strong novel findings, weak unmatched findings, malformed report shapes, and empty reports.
- [ ] Benchmark-specific project names and private finding content never appear in logs, fixtures, snapshots, or issue output.

### PR DESCRIPTION
**Pull request by aviggiano**
## Summary

- require terminal report issues to satisfy the canonical normalized finding schema before scoring
- allow the configured judge to promote reference matches that meet the score and confidence threshold
- reserve human review for supported novel findings, classify weak unsupported findings as false positives, and emit classification reason codes
- add focused regression coverage for report validation and each classification path

## Root cause

The terminal report contract accepted arbitrary issue objects, so scoring could receive findings without canonical matching fields. Separately, judge normalization preserved deterministic review classifications even for threshold-qualified reference matches, while title or summary text alone was enough to send unmatched findings to review.

## Verification

- pnpm -w test
- pnpm -w typecheck
- pnpm -w lint
- pnpm -w format:check

Closes #85

<h3>Greptile Summary</h3>

This PR fixes excessive routing of findings to the human review queue by introducing two interlocking changes: terminal report issues must now satisfy the canonical normalized finding schema before scoring proceeds, and classification is now fully derived by the caller rather than delegated to the LLM judge — the LLM only provides match scores and signals.

- **Report validation tightened**: `validateArtifactContract` for `ultrafuzz/report@1` now runs `validateFindingsSchema` on the `issues` array; a report whose issues lack the canonical fields (`schema_version`, `id`, `title`, `status`, `severity_guess`, `confidence`, `summary`) is rejected at the `readReport` boundary with `EVAL_TERMINAL_REPORT_INVALID`.
- **Classification policy moved to caller**: The LLM judge's `classification` field is removed from the schema and prompt; `normalizeLlmJudgeResult` derives classification deterministically using `judgeConfirmedMatch` (both score and confidence ≥ threshold) and `isStrongNovelFinding` (requires structural evidence), and the new `reason_code` field (`deterministic-match`, `judge-confirmed-match`, `strong-novel-finding`, `weak-unmatched-finding`) makes every routing decision auditable.
- **Evidence detection hardened**: `hasEvidence` is rewritten from a substring text-search (which matched words like \"latest\" containing \"test\") to structured field inspection (`proof_of_concept`, `poc`, `proof`, `reproduction`, `trace` top-level keys or `evidence` array entries with concrete values of ≥ 8 meaningful characters).

<h3>Confidence Score: 4/5</h3>

Safe to merge for the primary path; the candidate-mode + LLM-judge combination has a threshold inconsistency that should be addressed if that mode is used in production.

The classification overhaul is well-reasoned and correctly implemented for the default report mode. `normalizeLlmJudgeResult` gates LLM-promoted true-positives on `judgeScore >= input.threshold AND judgeConfidence >= input.threshold`, but the deterministic candidate-mode path uses `min(threshold, 0.45)` — if the LLM judge is ever paired with candidate mode, it cannot rescue findings that score 0.45–0.69 even though the deterministic path would. The default-mode evidence detection rewrite resolves the prior review-queue inflation from the text-search substring issue. The canonical schema gate in `validateArtifactContract` and `readReport` is tight and the rollback is tested.

`packages/evals/src/scoring.ts` around `normalizeLlmJudgeResult` — specifically the `judgeConfirmedMatch` threshold check and whether `matchMode` needs to be threaded into the `FindingJudge` input type if candidate-mode LLM judging is a real use case.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/evals/src/scoring.ts | Core classification logic refactored: LLM judge no longer owns classification; `normalizeLlmJudgeResult` derives it deterministically from `judgeConfirmedMatch` and `isStrongNovelFinding`; new `hasEvidence` replaces fragile text-search with structural field inspection. |
| packages/artifacts/src/artifact-contracts.ts | Adds `validateFindingsSchema` call on `issues` array inside `validateArtifactContract` for `ultrafuzz/report@1`; correctly returns failure when findings fail schema and includes no stale value in the error path. |
| packages/evals/src/types.ts | Adds `EvalClassificationReasonCode` union type and optional `reason_code?` on `FindingJudgeResult`; the optional marker correctly handles backward-compatibility with historical `scores.jsonl` artifacts. |
| packages/evals/test/scoring.test.ts | Comprehensive regression coverage added: canonical schema rejection, judge-confirmed promotion, weak-unmatched false-positive path, evidence-object edge cases, and empty-report all-missed case. |
| packages/artifacts/test/schema.test.ts | Adds two new test cases for the report contract: a fully canonical issue that should pass and an incomplete issue that should fail, covering the new `validateFindingsSchema` gate. |
| .ultrafuzz/prompts/review/final-report.md | Prompt updated to require canonical normalized finding schema fields and separates single-strategy `strategy` string from multi-strategy `strategy_provenance` object. |
| packages/evals/src/lineage.ts | Judge prompt version bumped v2 → v3 to track the new classification policy change in provenance fingerprints. |

</details>

<details><summary><h3>Flowchart</h3></summary>

<a href="#gh-light-mode-only">

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[Terminal report issues] --> B{validateFindingsSchema}
    B -- fail --> C[EVAL_TERMINAL_REPORT_INVALID]
    B -- pass --> D[scoreFindings]
    D --> E[bestMatch - deterministic]
    E --> F{score >= effectiveThreshold?}
    F -- yes --> G[deterministic-match true-positive]
    F -- no --> H{isStrongNovelFinding?}
    H -- yes --> I[needs-human-review strong-novel-finding]
    H -- no --> J[false-positive weak-unmatched-finding]
    G --> K{LLM judge enabled?}
    I --> K
    J --> K
    K -- no --> L[Use deterministic result]
    K -- yes --> M[normalizeLlmJudgeResult]
    M --> N{deterministicMatchedBugId?}
    N -- yes --> O[deterministic-match true-positive]
    N -- no --> P{judgeConfirmedMatch? score AND confidence >= threshold}
    P -- yes --> Q[judge-confirmed-match true-positive]
    P -- no --> R{isStrongNovelFinding?}
    R -- yes --> S[needs-human-review strong-novel-finding]
    R -- no --> T[false-positive weak-unmatched-finding]
    S --> U[Human Review Queue]
    Q --> V[True Positives]
    O --> V
    T --> W[False Positives]
```

</a>
<a href="#gh-dark-mode-only">

```mermaid
%%{init: {'theme': 'base', 'themeVariables': {"darkMode": true, "background": "#0d1117", "primaryColor": "#21262d", "primaryTextColor": "#e6edf3", "primaryBorderColor": "#8b949e", "lineColor": "#8b949e", "textColor": "#e6edf3", "edgeLabelBackground": "#161b22", "actorBkg": "#21262d", "actorBorder": "#8b949e", "actorTextColor": "#e6edf3", "actorLineColor": "#8b949e", "signalColor": "#8b949e", "signalTextColor": "#e6edf3", "noteBkgColor": "#373320", "noteBorderColor": "#d4a72c", "noteTextColor": "#f0e6c0", "labelBoxBkgColor": "#21262d", "labelBoxBorderColor": "#8b949e", "labelTextColor": "#e6edf3", "loopTextColor": "#e6edf3", "activationBkgColor": "#30363d", "activationBorderColor": "#8b949e"}}}%%
flowchart TD
    A[Terminal report issues] --> B{validateFindingsSchema}
    B -- fail --> C[EVAL_TERMINAL_REPORT_INVALID]
    B -- pass --> D[scoreFindings]
    D --> E[bestMatch - deterministic]
    E --> F{score >= effectiveThreshold?}
    F -- yes --> G[deterministic-match true-positive]
    F -- no --> H{isStrongNovelFinding?}
    H -- yes --> I[needs-human-review strong-novel-finding]
    H -- no --> J[false-positive weak-unmatched-finding]
    G --> K{LLM judge enabled?}
    I --> K
    J --> K
    K -- no --> L[Use deterministic result]
    K -- yes --> M[normalizeLlmJudgeResult]
    M --> N{deterministicMatchedBugId?}
    N -- yes --> O[deterministic-match true-positive]
    N -- no --> P{judgeConfirmedMatch? score AND confidence >= threshold}
    P -- yes --> Q[judge-confirmed-match true-positive]
    P -- no --> R{isStrongNovelFinding?}
    R -- yes --> S[needs-human-review strong-novel-finding]
    R -- no --> T[false-positive weak-unmatched-finding]
    S --> U[Human Review Queue]
    Q --> V[True Positives]
    O --> V
    T --> W[False Positives]
```

</a>
</details>

<details open><summary><h3>Comments Outside Diff (2)</h3></summary>

1. `packages/evals/src/scoring.ts`, line 1211-1217 ([link](https://github.com/monad-developers/ultrafuzz/blob/458c19710de208adc855c78e0b3e49e6ea7c679d/packages/evals/src/scoring.ts#L1211-L1217)) 

   <a href="#">P1</a> **`hasEvidence` text fallback: "test" matches common words as a substring**

   `hasEvidence` is now the load-bearing gate in `isStrongNovelFinding`, which determines whether an unmatched finding routes to human review or gets classified as a false positive. The text-fallback path uses `text.includes(term)` with `"test"` as a term, but `normalizeText` only lowercases and strips punctuation — it does not split on word boundaries. As a result, any finding whose JSON serialization contains a word like `"latest"`, `"contest"`, `"protest"`, or `"attestation"` will pass `hasEvidence` (all contain the substring `"test"`), get classified as `needs-human-review`, and enter the review queue when it should be a false positive. A finding with `title: "Uses latest block.timestamp"` reproduces this: `"latest".includes("test")` is `true` even though no evidence was submitted.

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: packages/evals/src/scoring.ts
   Line: 1211-1217

   Comment:
   **`hasEvidence` text fallback: "test" matches common words as a substring**

   `hasEvidence` is now the load-bearing gate in `isStrongNovelFinding`, which determines whether an unmatched finding routes to human review or gets classified as a false positive. The text-fallback path uses `text.includes(term)` with `"test"` as a term, but `normalizeText` only lowercases and strips punctuation — it does not split on word boundaries. As a result, any finding whose JSON serialization contains a word like `"latest"`, `"contest"`, `"protest"`, or `"attestation"` will pass `hasEvidence` (all contain the substring `"test"`), get classified as `needs-human-review`, and enter the review queue when it should be a false positive. A finding with `title: "Uses latest block.timestamp"` reproduces this: `"latest".includes("test")` is `true` even though no evidence was submitted.

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

   <a href="https://app.greptile.com/ide/claude-code?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fevals%2Fsrc%2Fscoring.ts%0ALine%3A%201211-1217%0A%0AComment%3A%0A**%60hasEvidence%60%20text%20fallback%3A%20%22test%22%20matches%20common%20words%20as%20a%20substring**%0A%0A%60hasEvidence%60%20is%20now%20the%20load-bearing%20gate%20in%20%60isStrongNovelFinding%60%2C%20which%20determines%20whether%20an%20unmatched%20finding%20routes%20to%20human%20review%20or%20gets%20classified%20as%20a%20false%20positive.%20The%20text-fallback%20path%20uses%20%60text.includes%28term%29%60%20with%20%60%22test%22%60%20as%20a%20term%2C%20but%20%60normalizeText%60%20only%20lowercases%20and%20strips%20punctuation%20%E2%80%94%20it%20does%20not%20split%20on%20word%20boundaries.%20As%20a%20result%2C%20any%20finding%20whose%20JSON%20serialization%20contains%20a%20word%20like%20%60%22latest%22%60%2C%20%60%22contest%22%60%2C%20%60%22protest%22%60%2C%20or%20%60%22attestation%22%60%20will%20pass%20%60hasEvidence%60%20%28all%20contain%20the%20substring%20%60%22test%22%60%29%2C%20get%20classified%20as%20%60needs-human-review%60%2C%20and%20enter%20the%20review%20queue%20when%20it%20should%20be%20a%20false%20positive.%20A%20finding%20with%20%60title%3A%20%22Uses%20latest%20block.timestamp%22%60%20reproduces%20this%3A%20%60%22latest%22.includes%28%22test%22%29%60%20is%20%60true%60%20even%20though%20no%20evidence%20was%20submitted.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=monad-developers%2Fultrafuzz&pr=96&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaude.svg?v=6">Fix in Claude Code</picture></a>

2. `packages/evals/src/types.ts`, line 308-321 ([link](https://github.com/monad-developers/ultrafuzz/blob/458c19710de208adc855c78e0b3e49e6ea7c679d/packages/evals/src/types.ts#L308-L321)) 

   <a href="#">P1</a> **`reason_code` is required but absent in all pre-existing serialized results**

   `reason_code` is a required field on `FindingJudgeResult` and is always present in newly produced results, but every `scores.jsonl` artifact written before this PR lacks it. Any code path that reads an old `scores.jsonl` (e.g., `compareEvalRun`, report rendering, or custom tooling that deserialises `EvalFindingScore`) will receive `undefined` where a `EvalClassificationReasonCode` string is expected. The judge prompt version bump (v2→v3) protects provenance-gated comparisons but does not prevent direct reads of previously written score files.

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: packages/evals/src/types.ts
   Line: 308-321

   Comment:
   **`reason_code` is required but absent in all pre-existing serialized results**

   `reason_code` is a required field on `FindingJudgeResult` and is always present in newly produced results, but every `scores.jsonl` artifact written before this PR lacks it. Any code path that reads an old `scores.jsonl` (e.g., `compareEvalRun`, report rendering, or custom tooling that deserialises `EvalFindingScore`) will receive `undefined` where a `EvalClassificationReasonCode` string is expected. The judge prompt version bump (v2→v3) protects provenance-gated comparisons but does not prevent direct reads of previously written score files.

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

   <a href="https://app.greptile.com/ide/claude-code?prompt=This%20is%20a%20comment%20left%20during%20a%20code%20review.%0APath%3A%20packages%2Fevals%2Fsrc%2Ftypes.ts%0ALine%3A%20308-321%0A%0AComment%3A%0A**%60reason_code%60%20is%20required%20but%20absent%20in%20all%20pre-existing%20serialized%20results**%0A%0A%60reason_code%60%20is%20a%20required%20field%20on%20%60FindingJudgeResult%60%20and%20is%20always%20present%20in%20newly%20produced%20results%2C%20but%20every%20%60scores.jsonl%60%20artifact%20written%20before%20this%20PR%20lacks%20it.%20Any%20code%20path%20that%20reads%20an%20old%20%60scores.jsonl%60%20%28e.g.%2C%20%60compareEvalRun%60%2C%20report%20rendering%2C%20or%20custom%20tooling%20that%20deserialises%20%60EvalFindingScore%60%29%20will%20receive%20%60undefined%60%20where%20a%20%60EvalClassificationReasonCode%60%20string%20is%20expected.%20The%20judge%20prompt%20version%20bump%20%28v2%E2%86%92v3%29%20protects%20provenance-gated%20comparisons%20but%20does%20not%20prevent%20direct%20reads%20of%20previously%20written%20score%20files.%0A%0AHow%20can%20I%20resolve%20this%3F%20If%20you%20propose%20a%20fix%2C%20please%20make%20it%20concise.&repo=monad-developers%2Fultrafuzz&pr=96&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixInClaude.svg?v=6">Fix in Claude Code</picture></a>
</details>

<a href="https://app.greptile.com/ide/claude-code?prompt=Fix%20the%20following%201%20code%20review%20issue.%20Work%20through%20them%20one%20at%20a%20time%2C%20proposing%20concise%20fixes.%0A%0A---%0A%0A%23%23%23%20Issue%201%20of%201%0Apackages%2Fevals%2Fsrc%2Fscoring.ts%3A939-940%0A**%60judgeConfirmedMatch%60%20ignores%20%60candidate%60%20mode%20effective%20threshold**%0A%0A%60bestMatch%60%20applies%20%60effectiveThreshold%20%3D%20Math.min%28threshold%2C%200.45%29%60%20when%20%60matchMode%20%3D%3D%3D%20%22candidate%22%60%2C%20so%20the%20deterministic%20path%20promotes%20any%20finding%20with%20score%20%E2%89%A5%200.45.%20When%20the%20deterministic%20path%20misses%20%28score%20%3C%200.45%29%20and%20the%20LLM%20is%20also%20active%2C%20%60judgeConfirmedMatch%60%20requires%20%60judgeScore%20%3E%3D%20input.threshold%60%20%28e.g.%200.7%29%20%E2%80%94%20not%20the%20effective%20threshold%20of%200.45.%20A%20finding%20that%20the%20deterministic%20path%20would%20have%20promoted%20at%200.45%20in%20candidate%20mode%20%28had%20it%20scored%20there%29%20cannot%20be%20rescued%20by%20an%20LLM%20score%20of%200.5%E2%80%930.69%2C%20even%20though%200.5%20clears%20the%20candidate-mode%20bar.%20The%20%60matchMode%60%20is%20not%20forwarded%20into%20the%20%60FindingJudge%60%20input%2C%20so%20there%20is%20no%20way%20for%20%60normalizeLlmJudgeResult%60%20to%20recover%20it.%0A%0A&repo=monad-developers%2Fultrafuzz&pr=96&platform=github"><picture><source media="(prefers-color-scheme: dark)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaudeDark.svg?v=6"><source media="(prefers-color-scheme: light)" srcset="https://greptile-static-assets.s3.amazonaws.com/badges/FixAllInClaude.svg?v=6">Fix All in Claude Code</picture></a>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 1 code review issue. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 1
packages/evals/src/scoring.ts:939-940
**`judgeConfirmedMatch` ignores `candidate` mode effective threshold**

`bestMatch` applies `effectiveThreshold = Math.min(threshold, 0.45)` when `matchMode === "candidate"`, so the deterministic path promotes any finding with score ≥ 0.45. When the deterministic path misses (score < 0.45) and the LLM is also active, `judgeConfirmedMatch` requires `judgeScore >= input.threshold` (e.g. 0.7) — not the effective threshold of 0.45. A finding that the deterministic path would have promoted at 0.45 in candidate mode (had it scored there) cannot be rescued by an LLM score of 0.5–0.69, even though 0.5 clears the candidate-mode bar. The `matchMode` is not forwarded into the `FindingJudge` input, so there is no way for `normalizeLlmJudgeResult` to recover it.


`````

</details>

<sub>Reviews (5): Last reviewed commit: ["Merge remote-tracking branch &#39;origin/mai..."](https://github.com/monad-developers/ultrafuzz/commit/9ea57e052f1f66e8200f8c5a4ab50a5dfd8b9ea9) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=45472236)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

**Comment by aviggiano** [comment]
Independent review complete. I verified the issue, PR diff, discussion, and checks, then pushed follow-up commit cbce0c0 to align the final-report prompt with the stricter report contract: production issues in report.json are now instructed and tested to satisfy the canonical normalized finding fields, so built-in report generation stays compatible with the new artifact validation.

Verification run:
- pnpm --filter @ultrafuzz/artifacts test
- pnpm --filter @ultrafuzz/prompts test
- pnpm --filter @ultrafuzz/evals test
- pnpm --filter @ultrafuzz/evals typecheck
- pnpm --filter @ultrafuzz/artifacts typecheck
- pnpm --filter @ultrafuzz/prompts typecheck
- pnpm -w test
- pnpm -w typecheck
- pnpm -w lint
- pnpm -w format:check

All PR checks are passing at head cbce0c0. No further blocking findings.

**Comment by aviggiano** [comment]
Second independent review complete: approved at head 4b5e9e3.

I pushed two follow-up commits that:
- replaced substring-based evidence detection with concrete field/value checks, including placeholder, short-value, compact-reference, and empty-report coverage;
- kept historical score artifacts compatible by making reason codes optional while continuing to emit them for new built-in results;
- removed the unused model classification output and applied score/confidence thresholds consistently;
- resolved the final-report strategy shape collision by keeping the canonical strategy string and moving structured details to strategy_provenance.

Focused package verification and the full workspace test, typecheck, lint, and format checks passed. All current PR checks pass, the branch is mergeable, and I found no remaining blocking or useful non-blocking recommendations.

**Comment by aviggiano** [comment]
Integration reconciliation after PRs #98 and #97: merged the latest main and combined the canonical normalized issue requirements from this PR with the landed property-provenance and dual-backend report contract. Final reports now retain canonical issue fields and strategy compatibility while also requiring the validated property provenance section. Artifact, eval, prompt, and runtime suites pass, along with workspace typecheck, lint, docs, formatting, and build.

File: packages/evals/src/scoring.ts [hunk 1]
```diff
@@ -926,18 +934,25 @@ function normalizeLlmJudgeResult(
       : undefined;
   const matchedBugId = deterministicMatchedBugId ?? judgeMatchedBugId;
   const judgeScore = roundMetric(data.score);
+  const judgeConfidence = roundMetric(data.confidence);
   const score = deterministicMatchedBugId === undefined ? judgeScore : input.deterministicResult.score;
-  const plausible = isPlausibleFinding(input.finding);
-  const deterministicReviewRequired = input.deterministicResult.classification === "needs-human-review";
-  let classification = data.classification;
+  const judgeConfirmedMatch =
+    judgeMatchedBugId !== undefined && judgeScore >= input.threshold && judgeConfidence >= input.threshold;
```
[the remark this answers is not in the document]
Comment (aviggiano) [reply packages/evals/src/scoring.ts:940]: Addressed by follow-up PR #102. Candidate mode now passes its effective threshold to the optional judge path and uses the same threshold for deterministic confidence; focused regression coverage locks the shared policy.
